### ISSUE TITLE
Please fix warnings when -Weffc++

### ISSUE THREAD

**Issue by sheerluck** [kaitai-io/kaitai_struct#652]
I set -Weffc++ in CMAKE_CXX_FLAGS_RELEASE and got bunch of warnings in my face :(

**Comment by GreyCat** [comment]
Can you clarify what exactly are you looking for? I can guess that it's probably related to C++, and it's probably related to codegen, not the runtime? For option itself, it's also not very clear: as far as I know, there are at least two implementations of that (one in gcc and one in clang), and they might be pretty different.

Is it just about #496, or something else?

**Comment by sheerluck** [comment]
yeah, about that and also about generating ctor with full member initializer list
